src/Makefile

```diff
@@ -172,9 +172,9 @@ ifeq ($(COMP),gcc)
 			CXXFLAGS += -m$(bits)
 			LDFLAGS += -m$(bits)
 		endif
-	else
-		CXXFLAGS += -m$(bits)
-		LDFLAGS += -m$(bits)
+		ifeq ($(OS),GNU/Linux)
```

Comment:
Why do you remove the `else` clause while the fix is in the `if` part? This could break builds on other architectures, so why not insert the fix before the `else` and leave the `else` and anything below it as it is instead?

Reply:
Removing -m$(bits) from CXXFLAGS and LDFLAGS does not break the build on other arches. For example, Debian removes it for all architectures: https://sources.debian.org/patches/stockfish/11-1/cxxflags.patch/

They are completely invalid if GCC isn't compiled with multilib support and are unnecessary if GCC is built with multilib support anyway.

Reply:
I strongly doubt the `-m$(bits)` was put there without reason (what about ancient GCC versions for instance) . If it ain't broke, don't fix it.

Reply:
But the issue is that it *is* broke if your gcc is built without multilib.

Reply:
Or is this about building 32bit releases on 64bit machines? If so, perhaps there is a way to set the flags only if they would be different from the default, so multilib support would only be required when it's really needed.

Reply:
You're right -- now I have it check if GCC has multilib before adding the flag.

Reply:
Perhaps you should replace the hardcoded `gcc` call into `$(CXX)`? It assures we're querying the same compiler we're actually using.

Reply:
Mhh ... suppose I run `make ARCH=x86-32 build` on a 64 bit machine without multilib. Will it now silently make a x86-64 bit build? Is there a way we can make it fail instead?

Reply:
> 
> 
> Perhaps you should replace the hardcoded `gcc` call with `$(CXX)`? It assures we're querying the same compiler we're actually using.

Notice how this is under the section for if we are using gcc, hence a hardcoded call to gcc should be okay here.

Reply:
> 
> 
> Mhh ... suppose I run `make ARCH=x86-32 build` on a 64 bit machine without multilib. Will it now silently make a x86-64 bit build? Is there a way we can make it fail instead?

I'm not really sure how this would work. We'd have to actually check the host arch against the selected arch I guess. Even in the state this is currently in, I think it's better than the previous behavior.